CHANGES/8540.bugfix.rst

```diff
@@ -0,0 +1,7 @@
+WSS: Fix heartbeat timeout logic -- by :user:`arcivanov`.
+
+The :py:meth:`~aiohttp.client_ws.ClientWebSocketResponse._pong_not_received` did not
+ terminate a :py:meth:`~aiohttp.client_ws.ClientWebSocketResponse.receive` operation.
+ This change causes `_pong_not_received` to feed the `reader` an error message, causing
+ pending `receive` to terminate and return the error message. The error message contains
+ the exception :py:class:`~aiohttp.client_exceptions.ServerTimeoutError`.
```

Comment:
```
/home/runner/work/aiohttp/aiohttp/docs/[towncrier-fragments]:209: WARNING: py:meth reference target not found: aiohttp.client_ws.ClientWebSocketResponse._pong_not_received
/home/runner/work/aiohttp/aiohttp/docs/[towncrier-fragments]:205: WARNING: py:meth reference target not found: aiohttp.client_ws.ClientWebSocketResponse.receive
/home/runner/work/aiohttp/aiohttp/docs/[towncrier-fragments]:205: WARNING: py:class reference target not found: aiohttp.client_exceptions.ServerTimeoutError
```

```suggestion
The :py:meth:`~aiohttp.ClientWebSocketResponse._pong_not_received` did not
 terminate a :py:meth:`~aiohttp.ClientWebSocketResponse.receive` operation.
 This change causes `_pong_not_received` to feed the `reader` an error message, causing
 pending `receive` to terminate and return the error message. The error message contains
 the exception :py:class:`~aiohttp.ServerTimeoutError`.
```

I'm not sure if this will fix the above, but let's give it a try.